be/src/exec/pipeline/nljoin/nljoin_probe_operator.cpp

```diff
@@ -271,21 +328,27 @@ ChunkPtr NLJoinProbeOperator::_permute_chunk(RuntimeState* state) {
     _probe_row_start = _probe_row_current;
     for (; _probe_row_current < _probe_chunk->num_rows(); ++_probe_row_current) {
         // Last build chunk must permute a chunk
-        if (_curr_build_chunk_index == _num_build_chunks() - 1 && _num_build_chunks() > 1) {
+        bool is_last_build_chunk = _curr_build_chunk_index == _num_build_chunks() - 1 && _num_build_chunks() > 1;
```

Comment:
!_num_build_chunks() .empty() &&  _curr_build_chunk_index == _num_build_chunks() - 1 

when num_build_chunks().empty() return true, can this operator behavior correctly,  do you have add such test caces?  the empty build side must be result of the sub-plan to avoid a EmptySet node.

Reply:
Yes, empty build side is covered. But it is handled outside this function.